src/core/store-data.ts

```diff
@@ -1,56 +1,223 @@
-import { Base64 } from "js-base64";
-import { ErrorForLogger, errorLogger, logger } from ".";
 import { MetricData } from "../interfaces";
 import { octokitForDataRepo } from "./octokit";
-import { storeDataStaggerTimeout } from "./store-data-stagger-timeout";
+import { ErrorForLogger } from "./error-logger";
 import { LogErrors } from "../shared/log-messages";
+import {
+  getOptionalEnvVar,
+  getOptionalNumericEnvVar,
+  getRequiredEnvVar,
+} from "../shared/get-env-var";
+
+export class StoreDataError extends ErrorForLogger {
+  name = "StoreDataError";
+
+  static noReadAccess(status: number): StoreDataError {
+    return new StoreDataError(
+      "error",
+      LogErrors.dataRepoNoReadAccess,
+      status.toString()
+    );
+  }
+
+  static noWriteAccess(status: number): StoreDataError {
+    return new StoreDataError(
+      "error",
+      LogErrors.dataRepoNoWriteAccess,
+      status.toString()
+    );
+  }
+}
+
+async function getHeadShas(
+  owner: string,
+  repo: string,
+  branch: string
+): Promise<{ commitSha: string; treeSha: string }> {
+  const {
+    data: [latestCommit],
+  } = await octokitForDataRepo.request("GET /repos/{owner}/{repo}/commits", {
+    owner,
+    repo,
+    sha: branch,
+    per_page: 1,
+  });
+
+  return {
+    commitSha: latestCommit.sha,
+    treeSha: latestCommit.commit.tree.sha,
+  };
+}
 
 export const storeData = async (metricData: MetricData[]) => {
-  const errored: string[] = [];
+  if (metricData.length === 0) {
+    // this is just a bug in our code and doesn't get a message in LogErrors
+    throw new Error("storeData must be called with at least one dataset");
+  }
+
+  const owner = getRequiredEnvVar("REPO_OWNER");
+  const repo = getRequiredEnvVar("REPO_NAME");
+  const repoPath = getRequiredEnvVar("REPO_PATH");
+  const targetBranch = getRequiredEnvVar("TARGET_BRANCH");
```

Comment:
lets change the variable names to something that shows they are for the data repository
```suggestion
  const dataRepoOwner = getRequiredEnvVar("REPO_OWNER");
  const dataRepoName = getRequiredEnvVar("REPO_NAME");
  const dataRepoPath = getRequiredEnvVar("REPO_PATH");
  const dataRepoTargetBranch = getRequiredEnvVar("TARGET_BRANCH");
```

Reply:
Fair enough - I mainly used `owner` and `repo` because it was less to type in the request arguments, but that's not a good reason to be confusing ^^